src/StreamJsonRpc/NerdbankMessagePackFormatter.ISerializationContextBuilder.cs

```diff
@@ -0,0 +1,241 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Immutable;
+using System.Collections.ObjectModel;
+using System.IO.Pipelines;
+using MessagePack.Formatters;
+using Nerdbank.MessagePack;
+using PolyType;
+using PolyType.Abstractions;
+using PolyType.ReflectionProvider;
+using StreamJsonRpc.Reflection;
+
+namespace StreamJsonRpc;
+
+/// <summary>
+/// Serializes JSON-RPC messages using MessagePack (a fast, compact binary format).
+/// </summary>
+/// <remarks>
+/// The MessagePack implementation used here comes from https://github.com/AArnott/Nerdbank.MessagePack.
+/// </remarks>
+public sealed partial class NerdbankMessagePackFormatter
+{
+    /// <summary>
+    /// Provides a builder interface for configuring the serialization context.
+    /// </summary>
+    public interface IFormatterContextBuilder
```

Comment:
Considering the set of exotic types grows over time, using an *interface* with a method for each exotic type is a recipe for unpleasant APIs for binary breaking changes in the future because we can't freely add methods to an interface after it ships.

Reply:
I got the interface vs class breaking change compat backwards in my head.

Reply:
Removed the interface, using the class in the configuration lambda